packages/core/saved-objects/core-saved-objects-migration-server-internal/src/zdt/utils/get_current_index.ts

```diff
@@ -7,16 +7,26 @@
  */
 
 import { escapeRegExp } from 'lodash';
-import type { FetchIndexResponse } from '../../actions';
+import type { Aliases } from '../../model/helpers';
+
+export const getCurrentIndex = ({
+  indices,
+  aliases,
+  indexPrefix,
+}: {
+  indices: string[];
+  aliases: Aliases;
+  indexPrefix: string;
+}): string | undefined => {
+  // if there is already a current alias pointing to an index, we reuse the index.
+  if (aliases[indexPrefix] && indices.includes(aliases[indexPrefix]!)) {
+    return aliases[indexPrefix];
+  }
```

Comment:
Changed the logic of `getCurrentIndex` to reuse any index that the current global alias (e.g `.kibana`) points to, as it was required to re-use the existing v2 index.